### ISSUE TITLE
Bad MAAS behavior when installing hardware and remote LB checks

### ISSUE THREAD

**Issue by BjoernT** [rcbops/rpc-openstack#1015]
##### What version (commit) do you encounter this issue on?

RPC11.1 and up

##### What type of issue is this (bug, feature, documentation, regression)?

Bug

##### What is the expected behavior?

We have changed the defaults of installing hardware monitoring checks to not installing any because
dell_check and hp_check is set to false. Additionally maas_scheme should be changed to https.
I would much rather prefer to enable both checks and add a dependency on ansible_system_vendor to install either the dell or hp checks

##### What is actually happening?

No hardware checks are rolled out and remote checks point to http rather then our default protocol HTTPS

##### How can we reproduce this issue?

Just install MAAS checks on a bare metal machine

##### How can we verify your issue is addressed?

It installs dell or HP checks depending on machine vendor
Also the remote checks are pointing to HTTPS now by default

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?

High prio

##### What is your evaluation on the importance of this issue, and why?

All kilo installation have no hardware monitoring

###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.1
- [ ] liberty-12.0
- [ ] kilo

**Comment by prometheanfire** [comment]
could be similar to #931

**Comment by prometheanfire** [comment]
to summarize the issue, this is a request that we selectively enable either the dell or hp checks based on inspecting system info

**Comment by mancdaz** [comment]
@prometheanfire @BjoernT @git-harry IMHO, this is neither a bug nor a prio 1. it is a prio 3 feature enhancement at best. I personally don't believe that it is at all onerous for the deployer to decide whether they have HP hardware or Dell hardware and run the appropriate playbook.

**Comment by prometheanfire** [comment]
@mancdaz I'm fine with reclassifying it, it's not like we don't have enough meetings where this can be brought up.

**Comment by mancdaz** [comment]
@promethanfire yes indeed, though I can't always make the meetings so wanted to note my thoughts here

**Comment by git-harry** [comment]
@mancdaz yes I would agree here. The deployer needs to set a single var to get the appropriate monitoring and it should be simple enough to add this as part of a deployment.

**Comment by BjoernT** [comment]
@git-harry  I disagree as usually, especially in the prio. This is came up after changing a default behavior and the playbooks still don't support mixed cluster with dell and hp hardware within the same control plane.

**Comment by git-harry** [comment]
@BjoernT there has never been any automatic hardware detection. In Juno there were separate playbooks for Dell and HP and the deployer was required to run the appropriate one. The rpc-openstack commit 4059eb63 is the first one I can find using dell_check and that was created more than a year ago. The lack of appropriate monitoring is easily addressed by setting the appropriate var.

We have never been made aware of a requirement to support mixed hardware environments. That said you are able to set vars against individual hosts and so that requirement can be met by the current mechanism.

**Comment by wesmattson** [comment]
I agree with @BjoernT on this. It seems like something that would be simple to solve for since ansible can detect the hardware and would automate an otherwise manual step that can easily be missed by the deployments team. Open Compute hardware is coming as well so there will be 3 vendors.

**Comment by shane-c** [comment]
+1 Can we get this bumped as a priority? 

I don't see why we would want to leave this up to the deployer to set a manual variable when ansible is collecting the system vendor in ansible facts and can use this to deploy the correct hardware checks based on that. 

OpenCompute hardware checks are a new feature request (https://github.com/rcbops/u-suk-dev/issues/241), adding even more complexity to this manual setting in a mixed vendor environment.

**Comment by git-harry** [comment]
@shane-c @wesmattson the weekly triage meetings, held on Mondays, are where we approve the issues the engineering team will work on next. If you wish to see this issue addressed sooner, please attend or speak with your representatives, so that it is considered at the next meeting.

**Comment by antonym** [comment]
You can now use host_check: true to automatically detect whether to install Dell or HP checks going foward in Newton:

https://github.com/rcbops/rpc-openstack/pull/1727
